api/v1alpha1/types_etcd.go

```diff
@@ -178,6 +178,13 @@ type BackupSpec struct {
 	// DeltaSnapshotMemoryLimit defines the memory limit after which delta snapshots will be taken
 	// +optional
 	DeltaSnapshotMemoryLimit *resource.Quantity `json:"deltaSnapshotMemoryLimit,omitempty"`
+	// DeltaSnapshotRetentionPeriod defines the duration for which delta snapshots will be retained, excluding the latest snapshot set.
+	// The value should be a string formatted as a duration (e.g., '1s', '2m', '3h', '4d')
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9][0-9]*([.][0-9]+)?(s|m|h|d))+$"
```

Comment:
@seshachalam-yv one more nit here. The current pattern allows values with multiple decimal points, such as `2.5.8.4d`, which is invalid. Please limit the number of decimal points to 0 or 1, with 1 or more digits following it only if the decimal point exists.